backends/platform/sdl/macosx/macosx_osys_misc.mm

```diff
@@ -0,0 +1,174 @@
+// https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/Cocoa64BitGuide/64BitChangesCocoa/64BitChangesCocoa.html
+#if __LP64__ || NS_BUILD_32_LIKE_64
+typedef unsigned long NSUInteger;
+#else
+typedef unsigned int NSUInteger;
+#endif
+
+// Those are not defined in the 10.4 SDK, but they are defined when targeting
+// Mac OS X 10.4 or above in the 10.5 SDK, and they do work with 10.4.
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_4
+enum {
+	NSUTF32StringEncoding = 0x8c000100,
+	NSUTF32BigEndianStringEncoding = 0x98000100,
+	NSUTF32LittleEndianStringEncoding = 0x9c000100
+};
+#endif
+#endif
+
+void OSystem_MacOSX::updateStartSettings(const Common::String & executable, Common::String &command, Common::StringMap &settings, Common::StringArray& additionalArgs) {
+	NSBundle* bundle = [NSBundle mainBundle];
+	// Check if scummvm is running from an app bundle
+	if (!bundle || ![bundle bundleIdentifier]) {
+		// Use default autostart implementation
+		OSystem_POSIX::updateStartSettings(executable, command, settings, additionalArgs);
+		return;
+	}
+
+	// If the bundle contains a scummvm.ini, use it as initial config
+	NSString *iniPath = [bundle pathForResource:@"scummvm" ofType:@"ini"];
+	if (iniPath && !settings.contains("initial-cfg"))
+		settings["initial-cfg"] = Common::String([iniPath fileSystemRepresentation]);
+
+	// If a command was specified on the command line, do not override it
+	if (!command.empty())
+		return;
+
+	// Check if we have an autorun file with additional arguments
+	NSString *autorunPath = [bundle pathForResource:@"scummvm-autorun" ofType:nil];
+	if (autorunPath) {
+		Common::File autorun;
+		Common::String line;
+		if (autorun.open(Common::FSNode([autorunPath fileSystemRepresentation]))) {
+			while (!autorun.eos()) {
+				line = autorun.readLine();
+				if (!line.empty() && line[0] != '#')
+					additionalArgs.push_back(line);
+			}
+		}
+		autorun.close();
+	}
+
+	// If the bundle contains a game directory, auto-detect it
+	NSString *gamePath = [[bundle resourcePath] stringByAppendingPathComponent:@"game"];
+	BOOL isDir = false;
+	BOOL exists = [[NSFileManager defaultManager] fileExistsAtPath:gamePath isDirectory:&isDir];
+	if (exists && isDir) {
+		// Ause auto-detection
+		command = "auto-detect";
+		settings["path"] = [gamePath fileSystemRepresentation];
+		return;
+	}
+
+	// The rest of the function has some commands executed only the first time after each version change
+	// Check the last version stored in the user settings.
+	NSString *versionString = [NSString stringWithUTF8String:gScummVMFullVersion];
+	NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
+	NSString *lastVersion = [defaults stringForKey:@"lastVersion"];
+	if (lastVersion && [lastVersion isEqualToString:versionString])
+		return;
+	[defaults setObject:versionString forKey:@"lastVersion"];
+
+	// If the bundle contains a games directory, add them to the launcher
```

Comment:
Does it make sense to perhaps share this code between ios/macosx?

Reply:
It probably does. The code is almost identical, with the only differences coming in the last commit for the sandboxed file paths.